reservation_units/models/reservation_unit.py

```diff
@@ -268,23 +273,231 @@ class Meta:
+                status=PricingStatus.PRICING_STATUS_ACTIVE,
+            )
+            .order_by("begins")
+            .values_list("pricing_type", flat=True)
+            .first()
+        )
 
-    @property
-    def reservation_state(self) -> ReservationState:
-        from reservation_units.utils.reservation_unit_reservation_state_helper import (
-            ReservationUnitReservationStateHelper,
+    @lookup_property
+    def publishing_state() -> str:
+        """State indicating the publishing status of the reservation unit."""
+        case = models.Case(
+            # Reservation Unit has been archived.
+            models.When(
+                models.Q(is_archived=True),
+                then=models.Value(ReservationUnitPublishingState.ARCHIVED.value),
+            ),
+            # Reservation Unit is still a draft.
+            models.When(
+                models.Q(is_draft=True),
+                then=models.Value(ReservationUnitPublishingState.DRAFT.value),
+            ),
+            # Reservation Unit is going to be published, but has not been set to unpublish in the future.
+            models.When(
+                (
+                    # Publishes in the future.
+                    models.Q(publish_begins__isnull=False)
+                    & models.Q(publish_begins__gt=Now())
+                    & (
+                        # Does not unpublish.
+                        models.Q(publish_ends__isnull=True)
+                        # Was previously unpublished.
+                        | models.Q(publish_ends__lte=Now())
+                        # Will be unpublished in the future, but before it re-publishes.
+                        | (
+                            models.Q(publish_ends__gt=Now())  #
+                            & models.Q(publish_begins__gt=models.F("publish_ends"))
+                        )
+                    )
+                ),
+                then=models.Value(ReservationUnitPublishingState.SCHEDULED_PUBLISHING.value),
+            ),
+            # Reservation Unit is currently unpublished or is going to be unpublished in the future.
+            models.When(
+                (
+                    (
+                        # Is unpublished.
+                        models.Q(publish_ends__isnull=False)
+                        & models.Q(publish_ends__lte=Now())
+                        & (
+                            # Was previously always published.
+                            models.Q(publish_begins__isnull=True)
+                            # Was previously published, but before it was unpublished.
+                            | (
+                                models.Q(publish_begins__lte=Now())  #
+                                & models.Q(publish_begins__lte=models.F("publish_ends"))
+                            )
+                        )
+                    )
+                    | (
+                        # Publishes and unpublishes at the exact same time in the future.
```

Comment:
Can be the same kind of zero length check (regardless of whether future or otherwise) as in `ReservationState.RESERVATION_CLOSED`:
```
models.Q(publish_begins__isnull=False)
& models.Q(publish_ends__isnull=False)
& models.Q(publish_begins=models.F("publish_ends"))
```

Reply:
Done.